gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -296,6 +296,14 @@ private static void modifyStateIfRetryRequired(org.apache.gobblin.configuration.
 
     mergedState.putAll(fallbackState.getProperties());
     mergedState.putAll(state.getProperties());
+    // Set CURRENT_ATTEMPTS_FIELD to right value to avoid continually retrying
+    if (fallbackState.getPropAsInt(TimingEvent.FlowEventConstants.CURRENT_ATTEMPTS_FIELD,0) > state.getPropAsInt(TimingEvent.FlowEventConstants.CURRENT_ATTEMPTS_FIELD,0)) {
```

Comment:
Why the default is 0, should not it be 1?

Reply:
In this case, we meet this condition and go into the "if" branch only when fallbackState has the prop and it's larger than 0 (at least 1). But if I set default to be one, in the situation that the fallback state does not have the prop and state has the prop set to be 0, we will update it which is unexpected

Reply:
in line 277 also the default is 1

Reply:
I see. It should work, but if the fallback state does not have the prop and state has the prop set to 0, the if condition will fail.
I think we should better make sure that the currentAttempt is never 0, it should be either not be set in the state, or set to 1 or higher